packages/vaadin-overlay/src/vaadin-overlay-position-mixin.js

```diff
@@ -80,25 +81,58 @@ export const PositionMixin = (superClass) =>
     static get observers() {
       return [
         '__positionSettingsChanged(positionTarget, horizontalAlign, verticalAlign, noHorizontalOverlap, noVerticalOverlap)',
-        '__overlayOpenedChanged(opened)',
+        '__overlayOpenedChanged(opened, positionTarget)',
       ];
     }
 
     constructor() {
       super();
 
-      this.__boundUpdatePosition = this._updatePosition.bind(this);
+      this._updatePosition = this._updatePosition.bind(this);
     }
 
-    __overlayOpenedChanged(opened) {
-      // Toggle the event listeners that cause the overlay to update its position
-      ['scroll', 'resize'].forEach((eventName) => {
-        if (opened) {
-          window.addEventListener(eventName, this.__boundUpdatePosition);
-        } else {
-          window.removeEventListener(eventName, this.__boundUpdatePosition);
-        }
+    connectedCallback() {
+      super.connectedCallback();
+
+      if (this.opened) {
+        this.__addUpdatePositionEventListeners();
+      }
+    }
+
+    disconnectedCallback() {
```

Comment:
nit: please add `/** @protected */` JSDoc annotation to lifecycle callbacks.

Reply:
Done.